src/main/scala/zio/s3/UploadOptions.scala

```diff
@@ -0,0 +1,42 @@
+package zio.s3
+
+import software.amazon.awssdk.services.s3.model.{ CreateMultipartUploadRequest, ObjectCannedACL, PutObjectRequest }
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * The options of the multipart upload and the put object request.
+ * @param metadata the user-defined metadata without the "x-amz-meta-" prefix
+ * @param cannedAcl a canned acl, defaults to "private"
+ * @param contentType the content type of the object (application/json, application/zip, text/plain, ...)
+ */
+case class UploadOptions(
+  metadata: Map[String, String] = Map.empty,
+  cannedAcl: ObjectCannedACL = ObjectCannedACL.PRIVATE,
+  contentType: Option[String] = None
+)
+
```

Comment:
```scala
case class UploadOptions(
  metadata: Map[String, String],
  cannedAcl: ObjectCannedACL
  contentType: Option[String]
)

object UploadOptions {
  val default: UploadOptions = MultipartUploadOptions(Map.empty, ChunkSize.default, ObjectCannedACL.PRIVATE)
}

sealed trait ChunkSize {
  def size: Int
}

object ChunkSize {
  //part size limit is 5Mb, required by amazon api
  val Min = 5 * 1024 * 1024

  val default = new ChunkSize {
    val size: Int = Min
  }

  def from(_size: Int): Either[String, ChunkSize] =
    _size match {
      case s if s >= Min =>
        Right(value = new ChunkSize {
          val size: Int = _size
        })
      case invalid       => Left(s"Invalid chunk size $invalid, minimum value is $Min ")
    }
}

case class MultipartUploadOptions(upload; UploadOptions, chunkSize: ChunkSize)

object MultipartUploadOptions {
  val default: MultipartUploadOptions = MultipartUploadOptions(UploadOptions.default, ChunkSize.default)
}

```

Reply:
looks good to me.
Can we keep the chunk size as Int? 
It would be easier to use because no additional steps except for the multipart upload call is needed. And as I mentioned above there is a dedicated S3Exception produced when the chunk size is outside the valid range.

Reply:
I would prefer to handle as a sealed trait, like this user would have to understand why this Either :)

Reply:
then the "parallelism" has to also be a type because it can be -1, for example. Obviously there is a check inside zio that doesn't allow negative parallelism, but zio's `mapMPar` still accepts a plain Int

Reply:
true !
you can validate this number inside the method :) if you want.
I let you decide. my choice would be do nothing ;p

Reply:
there is a dedicated config now https://github.com/zio/zio-s3/pull/117/commits/7f6632881537df013d77038a4e2f486a2a1dbb7a